src/extension.ts

```diff
@@ -62,9 +63,9 @@ async function openWindowsTerminal(profile: IWTProfile, uri?: vscode.Uri) {
       if (isWindows) {
         // remove first /
         const uriWindows = uri.path.substring(1, uri.path.length);
-        args.push('ssh', remoteMachine, `powershell -NoExit -Command cd ${uriWindows}\\;${vscode.env.shell}`);
+        args.push('ssh', remoteMachine, `powershell -NoExit -Command cd ${uriWindows}\\;${getShell(OS.WINDOWS)}`);
```

Comment:
vscode.env.shell is meant to do this, it is the resolved shell that launching the integrated terminal regularly would use. It should take into account $SHELL, fallbacks, the setting etc.

Reply:
Ah I see the problem in https://github.com/Tyriar/vscode-windows-terminal/issues/20#issuecomment-640857537, we're running as a UI extension so we can't leverage env.shell ☹️

Reply:
Yup vscode.env.shell  returns the shell from  "vscode ui" which will always be a windows shell since we need to run in windows.